Update dependencies & enable Dependabot version updates

Pl217 opened the pull request.

Pl217 committed:
Update ESLint to `v8.57.0`

Pl217 committed:
Update Typescript to `v5.5.4`

Pl217 committed:
Update `@unocha/hpc-repo-tools` to `v4.2.0`

Pl217 committed:
Upgrade `@types/node` to `v22.5.3`

Pl217 committed:
Upgrade Husky to `v9.1.5`

Migration guides in these two releases
https://github.com/typicode/husky/releases/tag/v9.0.1
https://github.com/typicode/husky/releases/tag/v9.1.1

Pl217 committed:
Update `lint-staged` to `v15.2.10`

Pl217 committed:
Update Prettier to `v3.3.3`

Pl217 committed:
Format files after updating Prettier to `v3.3.3`

This commit is the output of running command:
`yarn prettier -w .`

Pl217 committed:
Ignore revision with Prettier mass-formatting

Pl217 committed:
Update TS target to `es2023`

Pl217 committed:
Enable automatic Dependabot version updates

enxtur reviewed (changes requested):
It seems that only `src/db/models/authTarget.ts` has a linting issue, even though it hasn’t been modified. Should we consider applying linting checks to only to the touched files besides fix this issue?

Pl217 committed:
Fix linting error `unicorn/no-anonymous-default-export`

Since `@unocha/hpc-repo-tools` was updated from
`4.0.0` to `v4.2.0`, that also includes upgrade
in `v4.1.0` of `eslint-plugin-unicorn` library
from v49 to v52 and in `v52.0.0` of that library,
rule `no-anonymous-default-export` was added

https://github.com/UN-OCHA/hpc-repo-tools/releases/tag/v4.1.0
https://github.com/sindresorhus/eslint-plugin-unicorn/releases/tag/v52.0.0

Pl217 commented:
> It seems that only `src/db/models/authTarget.ts` has a linting issue, even though it hasn’t been modified. Should we consider applying linting checks to only to the touched files besides fix this issue?

I have fixed the linting error and updated packages which got patch version updates since opening of this PR (`@types/node` and `lint-staged`) to avoid Dependabot opening PRs to update them immediately after merging this PR, because version updates run monthly as configured, but also after each change to `dependabot.yml` file.

Regarding what we want to lint, checking linting and formatting in entire repository is exactly what we want actually. Because doing so caught a problem that wouldn't be caught if we check only changed files. In other repos, we check only changed files because there are existing problems in old legacy files which we didn't spend time resolve.
